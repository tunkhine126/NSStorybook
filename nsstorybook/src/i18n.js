/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 *   IMPORTANT: This file is used by the internal build
 *   script `extract-intl`, and must use CommonJS module syntax
 *   You CANNOT use import/export in this file.
 */
const addLocaleData = require('react-intl').addLocaleData; //eslint-disable-line
const enLocaleData = require('react-intl/locale-data/en');
const esLocaleData = require('react-intl/locale-data/es');
const ptLocaleData = require('react-intl/locale-data/pt');

// Custom Locale-Data for Haitian / Creole (not supported by react-intl)
const htLocaleData = {
  locale: 'ht',
  pluralRuleFunction: (e, t) => (t && e === 1 ? 'one' : 'other'),
};

const combineTranslationMessages = require('./translations/langs.json');

addLocaleData(enLocaleData);
addLocaleData(esLocaleData);
addLocaleData(ptLocaleData);
addLocaleData(htLocaleData);

const DEFAULT_LOCALE = 'en';

// prettier-ignore
const appLocales = [
  'en',
  'es',
  'pt',
  'ht'
];

const formatTranslationMessages = (locale, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE
      ? formatTranslationMessages(DEFAULT_LOCALE, combineTranslationMessages)
      : {};

  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE
        ? defaultFormattedMessages[key]
        : messages[key];
    return Object.assign(formattedMessages, {
      [key]: formattedMessage[locale],
    });
  };
  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

const translationMessages = {
  en: formatTranslationMessages('en', combineTranslationMessages),
  es: formatTranslationMessages('es', combineTranslationMessages),
  pt: formatTranslationMessages('pt', combineTranslationMessages),
  ht: formatTranslationMessages('ht', combineTranslationMessages),
};

exports.appLocales = appLocales;
exports.formatTranslationMessages = formatTranslationMessages;
exports.translationMessages = translationMessages;
exports.DEFAULT_LOCALE = DEFAULT_LOCALE;
