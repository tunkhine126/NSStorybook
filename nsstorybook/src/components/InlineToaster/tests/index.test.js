/**
 *
 * Tests for InlineToaser
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { DEFAULT_LOCALE } from '../../../../i18n';
import InlineToaser from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('<InlineToaser />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <InlineToaser clearToaster={() => {}} variant="success" />
      </IntlProvider>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should hover over toaster then hover out', () => {
    const spy = jest.spyOn(global.console, 'error');

    const wrapper = mount(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <InlineToaser
          clearToaster={() => {}}
          variant="success"
          timer={0}
          message="test"
        />
      </IntlProvider>
    );

    const toaster = wrapper.find('.inline-toaster').first();

    toaster.simulate('mouseenter');
    toaster.simulate('mouseleave');

    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <InlineToaser clearToaster={() => {}} variant="success" />
      </IntlProvider>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
