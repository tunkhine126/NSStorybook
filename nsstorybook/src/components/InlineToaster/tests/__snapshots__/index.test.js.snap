// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineToaser /> Should render and match the snapshot 1`] = `
<span
  class="inline-toaster styles__InlineToasterWrapper-sc-11dlxly-0 hCoZQG"
>
  <div
    aria-describedby="toaster"
    class="MuiPaper-root MuiSnackbarContent-root makeStyles-success-49 MuiPaper-elevation6"
    role="alert"
  >
    <div
      class="MuiSnackbarContent-message"
    >
      <span
        class="makeStyles-message-55"
        id="toaster"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-53 makeStyles-iconVariant-54"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
      </span>
    </div>
  </div>
</span>
`;
